go/marshal/encode.go

```diff
@@ -18,27 +20,33 @@ import (
 
 // Marshal converts a Go value to a Noms value.
 //
-// Marshal traverses the value v recursively. Marshal uses the following type-dependent encodings:
+// Marshal traverses the value v recursively. Marshal uses the following
+// type-dependent encodings:
 //
 // Boolean values are encoded as Noms types.Bool.
 //
-// Floating point and integer values are encoded as Noms types.Number. At the moment this might lead to some loss in precision because types.Number currently takes a float64.
+// Floating point and integer values are encoded as Noms types.Number. At the
+// moment this might lead to some loss in precision because types.Number
+// currently takes a float64.
 //
 // String values are encoded as Noms types.String.
 //
 // Slices and arrays are encoded as Noms types.List.
 //
-// Maps are encoded as Noms types.Map
+// Maps are encoded as Noms types.Map, or if the value is struct{}, a types.Set.
```

Comment:
Oops yeah, comment left over from the half finished patch which implemented this anyway.